test/fixtures/azure-fixture/src/main/java/fixture/azure/MockAzureBlobStore.java

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package fixture.azure;
+
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.bytes.CompositeBytesReference;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.rest.RestStatus;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class MockAzureBlobStore {
+
+    private static final Logger logger = LogManager.getLogger(MockAzureBlobStore.class);
+    private static final BytesReference EMPTY = new BytesArray(new byte[] {});
+
+    public enum BlobType {
+        BLOCK("BlockBlob") {
+            @Override
+            AzureBlob create() {
+                return new MockAzureBlockBlob();
+            }
+        },
+        PAGE("PageBlob") {
+            @Override
+            AzureBlob create() {
+                throw new UnsupportedOperationException("PageBlob is not supported");
+            }
+        },
+        APPEND("AppendBlob") {
+            @Override
+            AzureBlob create() {
+                throw new UnsupportedOperationException("AppendBlob is not supported");
+            }
+        };
+
+        final String xMsBlobType;
+
+        BlobType(String xMsBlobType) {
+            this.xMsBlobType = xMsBlobType;
+        }
+
+        abstract AzureBlob create();
+
+        public String getXMsBlobType() {
+            return xMsBlobType;
+        }
+
+        public static BlobType fromXMSBlobType(String blobTypeString) {
+            for (BlobType blobType : BlobType.values()) {
+                if (blobType.xMsBlobType.equals(blobTypeString)) {
+                    return blobType;
+                }
+            }
+            return null;
+        }
+    }
+
+    private final Map<String, AzureBlob> blobs;
+
+    public MockAzureBlobStore() {
+        this.blobs = new ConcurrentHashMap<>();
+    }
+
+    public void putBlock(String path, String blockId, BytesReference content, String leaseId) {
```

Comment:
This might be a bit anonying suggestion. But can we put `@Nullable` annotation to the leaseId parameters where it can be `null`? I feel it would be helpful for readers.

Reply:
No, totally agree. Addressed in 74fb420a1d81525bc7998253a58bf2f57d6548e9